processor/src/main/java/org/mapstruct/ap/spi/DefaultAccessorNamingStrategy.java

```diff
@@ -55,7 +55,18 @@ else if ( isPresenceCheckMethod( method ) ) {
         }
     }
 
-    private boolean isGetterMethod(ExecutableElement method) {
+    /**
+     * Returns true when the {@link ExecutableElement} is a getter method. A getter is a getter when it starts with
+     * 'get' and the return type is any type other than void OR the getter starts with 'is' and the type returned is
+     * a primitive or the wrapper for boolean {@link Boolean}. NOTE: the latter does strictly not comply to the bean
+     * convention. The remainder of the name is supposed to reflect the property name.
+     *
+     * MapStruct will enforce that there's no argument.
```

Comment:
The calling code by itself is also a bit cryptic I guess. Perhaps: MapStruct guarantees that the given method has no arguments.